### PR TITLE
docs: fix links from TOC to Headings and improve appearance of docs (a little)

### PR DESCRIPTION
**Pull request by JohnMcLear**
Resolves https://github.com/ether/etherpad-lite/issues/4843

This code is flakey tho and lots of it is just ignored, at least this resolves the links but the docs generation in general needs some attention.

### CODE REVIEW

File: src/bin/doc/package-lock.json [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-  "lockfileVersion": 1,
-  "requires": true,
-  "dependencies": {
-    "marked": {
-      "version": "1.1.1",
-      "resolved": "https://registry.npmjs.org/marked/-/marked-1.1.1.tgz",
-      "integrity": "sha512-mJzT8D2yPxoPh7h0UXkB+dBj4FykPJ2OIfxAWeIHrvoHDkFxukV/29QxoFQoPM6RLEwhIFdJpmKBlqVM3s2ZIw=="
-    }
-  }
-}
```
Comment (rhansen) [inline src/bin/doc/package-lock.json:13]: Why was this file deleted?

File: src/bin/doc/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "node": ">=10.17.0"
   },
   "dependencies": {
-    "marked": "1.1.1"
+    "marked": "2.0.0"
```
Comment (rhansen) [inline src/bin/doc/package.json:10]:
```suggestion
    "marked": "^2.0.0"
```

**Review by rhansen — APPROVED** [review]

File: src/bin/doc/package-lock.json (hunk 1 above)
Comment (JohnMcLear) [reply src/bin/doc/package-lock.json:13]: Weird that it removed it..  It should have re-added it when doing ``npm i``
